### PR TITLE
fix: theme is not configured in app router

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4336, CAL-4436
- [Loom](https://www.loom.com/share/d2d1b3969f7842618b7397942de7860d?sid=78e50ecc-10fe-4bb4-b00e-ad141d35e1a3)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/playwright/change-theme.e2e.ts [hunk 1]
```diff
@@ -1,14 +1,18 @@
 import { expect } from "@playwright/test";
+import { testBothFutureAndLegacyRoutes } from "playwright/lib/future-legacy-routes";
 
 import { test } from "./lib/fixtures";
 
-test.describe("Change Theme Test", () => {
-  test("change theme to dark", async ({ page, users }) => {
-    const pro = await users.create();
+testBothFutureAndLegacyRoutes.describe("Change Theme Test", () => {
```
Comment (hbjORbj) [inline apps/web/playwright/change-theme.e2e.ts:6]: adding `testBothFutureAndLegacyRoutes` to test app router pages

File: apps/web/playwright/change-theme.e2e.ts [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+  const username = "prousername";
+  test.beforeEach(async ({ page, users }) => {
```
Comment (hbjORbj) [inline apps/web/playwright/change-theme.e2e.ts:8]: small refactor to reduce code

File: apps/web/playwright/change-theme.e2e.ts (hunk 1 above)
Comment (hbjORbj) [reply apps/web/playwright/change-theme.e2e.ts:6]: Can you turn on `APP_ROUTER_SETTINGS_MY_ACCOUNT_ENABLED` to `1` for dev

File: apps/web/app/layoutHOC.tsx [hunk 3]
```diff
@@ -51,7 +51,7 @@ export function WithLayout<T extends Record<string, any>>({
         requiresLicense={requiresLicense || !!(Page && "requiresLicense" in Page && Page.requiresLicense)}
         nonce={nonce}
         themeBasis={null}
-        isThemeSupported={!!(Page && "isThemeSupported" in Page && Page.isThemeSupported)}
+        isThemeSupported={Page && "isThemeSupported" in Page ? (Page.isThemeSupported as boolean) : undefined}
```
Comment (sean-brydon) [inline apps/web/app/layoutHOC.tsx:54]: NIT: I think `isThemeSupported={Page?.isThemeSupported ?? undefined}` has the same effect here and is a lot more readable 
```suggestion
        isThemeSupported=isThemeSupported={Page?.isThemeSupported ?? undefined}
) : undefined}
```

**Review by sean-brydon — APPROVED** [review]
LGTM - happy merging but i'd personally like to see the readability of that tenery improved

File: apps/web/app/layoutHOC.tsx (hunk 3 above)
Comment (hbjORbj) [reply apps/web/app/layoutHOC.tsx:54]: `Page?.isThemeSupported` will throw type error :( Feel free to make tweaks

**Pull request merged**